kornia/augmentation/_2d/intensity/random_snow.py

```diff
@@ -0,0 +1,60 @@
+from typing import Any, Dict, Optional
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.color import hls_to_rgb, rgb_to_hls
+from kornia.core import Tensor
+
+
+class RandomSnow(IntensityAugmentationBase2D):
+    r"""Generates snow effect on given tensor image or a batch tensor images.
+
+    Args:
+        snow_coefficient (float): A float between 0 and 1 that controls the amount of snow to add to the
+        image, the larger value corresponds to the more snow.
+        brightness (float): A float greater than 1 that controls the brightness of the snow.
+        same_on_batch (bool): If True, apply the same transformation to each image in a batch. Default: False.
+        p (float): Probability of applying the transformation. Default: 0.5.
+        keepdim (bool): Keep the output tensor with the same shape as input. Default: False.
+
+    Returns:
+        torch.Tensor: The snowed image with shape (C, H, W).
+
+    Examples:
+        >>> import torch
+        >>> import kornia
+        >>> rng = torch.manual_seed(0)
+        >>> inputs = torch.rand(2, 3, 4, 4)
+        >>> snow = kornia.augmentation.RandomSnow(p=1.0, snow_coefficient=0.2, brightness=2.0)
+        >>> output = snow(inputs)
+        >>> output.shape
+        torch.Size([2, 3, 4, 4])
+    """
+
+    def __init__(
+        self,
+        snow_coefficient: float = 0.5,
+        brightness: float = 2.5,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, keepdim=keepdim)
+        if snow_coefficient < 0 or snow_coefficient > 1:
+            raise ValueError("Snow coefficient must be between 0 and 1.")
```

Comment:
from kornia.core.check
```suggestion
        KORNIA_CHECK(0 < snow_coefficient < 1, "Snow coefficient must be between 0 and 1")
```